src/backend/api/meals.js

```diff
@@ -2,13 +2,188 @@ const express = require("express");
+    }
+    
+    //Returns all meals that partially match the given title.
+    //Rød grød will match the meal with the title Rød grød med fløde. (api/meals?title=Rød%20grød)
+    if (title) {
+      const result = await knex
+        .select("*")
+        .from("meal")
+        .where("title", "like", `%${title}%`);
+
+      response.data = result;
+      response.status = 200;
+      response.message = "Meal name match found";
+    }
+
+    //Returns all meals where the date for meal_time is after the given date.(api/meals?dateAfter=2023-01-01)
+    if (dateAfter) {
+      console.log({ dateAfter, formatted: new Date(dateAfter) }); //meal_time is in datetime format in DB
+      const result = await knex
+        .select("*")
+        .from("meal")
+        .where("meal_time", ">", new Date(dateAfter));
+
+      response.data = result;
+      response.status = 200;
+      response.message = "Meals with date after the given date";
+    }
+
+    //Returns all meals where the date for meal_time is before the given date.(api/meals?dateBefore=2024-01-01)
+    if (dateBefore) {
+      console.log({ dateBefore, formatted: new Date(dateBefore) }); //meal_time is in datetime format in DB
+      const result = await knex
+        .select("*")
+        .from("meal")
+        .where("meal_time", "<", new Date(dateBefore));
+
+      response.data = result;
+      response.status = 200;
+      response.message = "Meals with date before the given date";
+    }
+
+    //Returns the given number of meals (api/meals?limit=7)
+    if (limit) {
+      const result = await knex
+        .select("*")
+        .from("meal")
+        .orderBy("id", "asc")
+        .limit(+limit);
+
+      response.data = result;
+      response.status = 200;
+      response.message = `Here are the first ${limit} meals`;
+    }
+
+    //Returns all meals sorted by the given key. Allows when, max_reservations and price as keys.
+    //Default sorting order is asc(ending). (api/meals?sortKey=price)
+    if (["meal_time", "max_reservations", "price"].includes(sortKey)) {
+      const result = await knex
+        .select("*")
+        .from("meal")
+        .orderBy(sortKey, sortDir === "desc" ? sortDir : "asc");
+
+      response.data = result;
+      response.status = 200;
+      response.message = "Meals sorted by the given sortKey";
+    }
+
+    //Returns all meals sorted in the given direction.
+    //Only works combined with the sortKey and allows asc or desc. (api/meals?sortKey=price&sortDir=desc)
+    if (["meal_time", "max_reservations", "price"].includes(sortKey)) {
+      if (sortDir === "asc" || sortDir === "desc") {
+        const result = await knex
+          .select("*")
+          .from("meal")
+          .orderBy(sortKey, sortDir);
+
+        response.data = result;
+        response.status = 200;
+        response.message =
+          "Meals sorted by the given sort key and sort direction";
+      }
+    }
```

Comment:
This `if` statement is necessary? From what I see the previous `if` statement does the same thing, since it uses both `sortDir` and `sortKey` parameters as well.